code/modules/reagents/chemistry_reagents/medical.dm

```diff
@@ -685,12 +685,26 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/M, alien)
 	custom_metabolism = 0.2
 	overdose_threshold = REAGENTS_OVERDOSE/5
 	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL/5
-
-/datum/reagent/medicine/hyperzine/on_mob_add(mob/living/M)
-	M.reagent_move_delay_modifier -= 0.5
+	var/duration = 0
+
+/datum/reagent/medicine/hyperzine/on_mob_delete(mob/living/M)
+	var/amount = M.adjustOxyLoss(duration * 4)
+	M.adjustHalLoss(amount)
+	if(amount > 50)
+		M.KnockOut(amount * 0.1)
+		to_chat(M, "<span class='danger'>Your world convulses as a wave of extreme fatigue washes over you!</span>") //when hyperzine is removed from the body, there's a backlash as it struggles to transition and operate without the drug
+	else
+		M.KnockDown(amount * 0.05)
+		to_chat(M, "<span class='warning'>A sudden wave of fatigue washes over you.</span>")
+	if(M.stat == DEAD)
```

Comment:
This should be checked before anything, as it's pointless to apply knock downs/outs and messages on someone that already dropped dead from a spike of oxy loss.